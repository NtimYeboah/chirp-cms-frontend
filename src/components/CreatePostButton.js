/**
 * Created by ntimobedyeboah on 4/16/17.
 */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class CreatePostButton extends Component {
    render() {
        return (
            <div className="columns">
                <div className="column is-12">
                    <nav>
                        <div className="container block">
                            <div className="level-right">
                                <Link className="button is-primary" to="/posts/create">Add A Post</Link>
                            </div>
                        </div>
                    </nav>
                </div>
            </div>

        )
    }
}

export default CreatePostButton;