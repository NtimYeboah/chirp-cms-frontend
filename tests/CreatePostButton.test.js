/**
 * Created by ntimobedyeboah on 4/19/17.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { MemoryRouter, Link } from 'react-router-dom';

import CreatePostButton from './../src/components/CreatePostButton';

describe('CreatePostButton test', () => {
   test('it should render component and take snapshot', () => {
       const component = renderer.create(<MemoryRouter><CreatePostButton /></MemoryRouter>);

       let tree = component.toJSON();
       expect(tree).toMatchSnapshot();
   });
});

describe('CreatePostButton rendered output', () => {
   test('it should a link', () => {
       const createPostButton = mount(<MemoryRouter><CreatePostButton /></MemoryRouter>);

       expect(createPostButton.find(Link).text()).toEqual('Add A Post');
   });
});