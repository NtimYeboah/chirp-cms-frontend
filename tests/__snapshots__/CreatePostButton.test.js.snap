exports[`CreatePostButton test it should render component and take snapshot 1`] = `
<div
  className="columns">
  <div
    className="column is-12">
    <nav>
      <div
        className="container block">
        <div
          className="level-right">
          <a
            className="button is-primary"
            href="/posts/create"
            onClick={[Function]}>
            Add A Post
          </a>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
